File: backend/src/main/java/com/main/Difficult/Domain/producto/simple/Piso.java
package com.main.Difficult.Domain.producto.simple;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;
@Getter @Setter
@Document("Producto")
@TypeAlias("Piso")
public class Piso extends ProductoSimple {
    @JsonProperty("medidas_x") private Double medidasX = 0.00;
    @JsonProperty("medidas_y") private Double medidasY = 0.00;
    private Transito transito = Transito.NORMAL;
    private Terminacion terminacion = Terminacion.BRILLANTE;

    public Piso(String imagen, String nombre, String descripcion, Integer puntaje, String paisDeOrigen, Double precioBase, Double medidasX,Double medidasY, Transito transito, Terminacion terminacion) {
        super(imagen, nombre, descripcion, puntaje, paisDeOrigen, precioBase);
        this.setMedidasX(medidasX);
        this.setMedidasY(medidasY);
        this.setTransito(transito);
        this.setTerminacion(terminacion);
    }

    @Override
    public Double recargo() {
        return transito.getRecargo();
    }
}

File: backend/src/main/kotlin/aplicacion/dominio/producto/simple/Piso.kt
package aplicacion.dominio.producto.simple

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonView
import aplicacion.dominio.producto.View
import javax.persistence.Entity

@Entity
class Piso : ProductoSimple() {
    @JsonProperty("medidas_x") var medidasX: Double = 0.00
    @JsonProperty("medidas_y") var medidasY: Double = 0.00
    @JsonView(View.ProductoIndividual::class) var transito: Transito = Transito.NORMAL
    @JsonView(View.ProductoIndividual::class) var terminacion: Terminacion = Terminacion.BRILLANTE

    override fun recargo(): Double {
        return transito.recargo()
    }
}

enum class Transito {
    ALTO_TRANSITO {
        override fun recargo(): Double = 1.20
    },
    NORMAL {
        override fun recargo(): Double = 1.00
    };

    abstract fun recargo(): Double
}

enum class Terminacion {
    SATINADO, SEMI_SATINADO, BRILLANTE
}